src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,384 @@
+    vxlan_subnet=None,
+    validate_only=False,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given SDDC exists for given organization.
+
+    name
+        The name of SDDC that will be assigned to new created SDDC.
+
+    hostname
+        The host name of VMC.
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console.
+
+    org_id
+        The ID of organization to which the SDDC belongs to.
+
+    num_hosts
+        An integer value to indicate number of hosts in a SDDC.
+
+    provider: String
+        Determines what additional properties are available based on cloud provider.
+        Possible values are: AWS, ZEROCLOUD
+
+    region: String
+        AWS region on which SDDC will be deployed.
+
+    account_link_config
+        (Optional) The account linking configuration, we will keep this one and remove accountLinkSddcConfig finally.
+
+        account_link_config expects to be passed as a dict having delay_account_link parameter
+            delay_account_link: Boolean
+            (Optional) Boolean flag identifying whether account linking should be delayed or not for the SDDC.
+
+            .. code::
+
+                account_link_config
+                    - delay_account_link: False
+
+    account_link_sddc_config
+        (Optional) A list of the AccountLinkSddcConfig which indicates SDDC linking configurations to use.
+
+        AccountLinkSddcConfig has two parameters connected_account_id  and customer_subnet_ids
+
+        'connected_account_id': String
+            (Optional) The ID of the customer connected account to work with.
+
+        'customer_subnet_ids': Array of String
+            (Optional) List of subnet Ids
+
+    deployment_type: String
+        (Optional) Denotes if request is for a SingleAZ or a MultiAZ SDDC. Default is SingleAZ.
+        Possible values are: SingleAZ, MultiAZ
+
+    host_instance_type: String
+        (Optional) The instance type for the ESX hosts in the primary cluster of the SDDC.
+        Possible values are: i3.metal, r5.metal, i3en.metal
+
+    msft_license_config :
+        (Optional) Indicates the desired licensing support, if any, of Microsoft software.
+        It can be specified in the below format
+
+        .. code::
+
+            msft_license_config
+                - academic_license: False
+                - mssql_licensing: "DISABLED"
+                - windows_licensing: "DISABLED"
+
+             where,
+             academic_license - Flag to identify if it is Academic Standard or Commercial Standard License.
+             mssql_licensing - The status MSSQL licensing for this SDDC’s clusters.
+                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
+             windows_licensing - The status of Windows licensing for this SDDC’s clusters. Can be enabled, disabled, or customer’s.
```

Comment:
The `Possible values` below seems to make `Can be` redundant.

Reply:
Need to mention seprately , as we are talking about two different parameters here

Reply:
```
             windows_licensing - The status of Windows licensing for this SDDC’s clusters. Can be enabled, disabled, or customer’s.                                                           
                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
```

should be

```
             windows_licensing - The status of Windows licensing for this SDDC’s clusters.                                             
                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
```

Reply:
Updated in proper yaml codeblock